### PR TITLE
Header

### PR DESCRIPTION
**Pull request by Erwan60**

### CODE REVIEW

**Review by vingtle — APPROVED** [review]

File: client/src/Router.jsx [hunk 1]
```diff
@@ -1,11 +1,31 @@
 import { createBrowserRouter } from "react-router-dom";
 import Welcome from "./pages/Welcome/Welcome";
+import Home from "./pages/Home/Home";
+import Archive from "./pages/Archive/Archive";
+import Member from "./pages/Member/Menber";
```
Comment (VinchC) [inline client/src/Router.jsx]: Member/Member

File: client/src/components/Header/Header.jsx [hunk 2]
```diff
@@ -1 +1,92 @@
...
+  };
+
+  const togglePopover = () => {
+    setShowPopover(!showPopover);
+  };
+
+  return (
+    <header>
+      <div id="header">
+        <div className="logo-acceuil">
```
Comment (VinchC) [inline client/src/components/Header/Header.jsx]: name of the classes in English please

File: client/src/components/Header/Header.jsx [hunk 3]
```diff
@@ -1 +1,92 @@
...
+          className="img-container"
+          onKeyDown={togglePopover}
+          role="presentation"
+          onClick={togglePopover}
+        >
+          <img src={profil} alt="profil" className="profil" />
+          {/* Popover */}
+          {showPopover && (
+            <div className="popover">
+              <div className="popoverEcrit">
```
Comment (VinchC) [inline client/src/components/Header/Header.jsx]: className in English

File: client/src/components/Header/Header.jsx [hunk 4]
```diff
@@ -1 +1,92 @@
...
+                <ul>
+                  <li>Compte</li>
+                  <div className="popoverMailImg">
+                    <li>
+                      <img src={profil} alt="profil" className="profil" />
```
Comment (VinchC) [inline client/src/components/Header/Header.jsx]: profile

File: client/src/components/Header/Header.jsx [hunk 5]
```diff
@@ -1 +1,92 @@
...
+                    </li>
+                    <div className="popoverNom">
```
Comment (VinchC) [inline client/src/components/Header/Header.jsx]: className in English

File: client/src/components/Header/header.css [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+}
+
+.logo {
+  width: 100px;
+  cursor: pointer;
+}
+.profil {
```
Comment (VinchC) [inline client/src/components/Header/header.css]: profile

File: client/src/components/Header/header.css [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+  border-radius: 14px;
+  color: #e3e3e3;
+  background-color: #6499ff;
+  width: 100px;
+  cursor: pointer;
+  margin-left: 150px;
+  padding: 10px 30px;
+}
+
+.logo-acceuil {
```
Comment (VinchC) [inline client/src/components/Header/header.css]: English

File: client/src/components/Header/header.css [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+.popover {
+  position: absolute;
+  background-color: #2c2c2c;
+  width: 300px;
+  padding-bottom: 50px;
+  right: 2.5px;
+  border-radius: 20px;
+}
+
+.popoverEcrit {
```
Comment (VinchC) [inline client/src/components/Header/header.css]: English

File: client/src/components/Header/header.css [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+  display: flex;
+  justify-content: space-around;
+}
+
+.popoverMailImg {
+  display: flex;
+  flex-direction: row;
+}
+
+.popoverNom {
```
Comment (VinchC) [inline client/src/components/Header/header.css]: English

File: client/src/components/Header/header.css [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+}
+
+.popoverEcrit ul {
```
Comment (VinchC) [inline client/src/components/Header/header.css]: English

File: client/src/main.jsx [hunk 11]
```diff
@@ -1,11 +1,44 @@
...
 
-ReactDOM.createRoot(document.getElementById('root')).render(
+import React from "react";
+import ReactDOM from "react-dom/client";
+
+import { createBrowserRouter, RouterProvider } from "react-router-dom";
+
+import Home from "./pages/Home/Home";
+import Archive from "./pages/Archive/Archive";
+import Member from "./pages/Member/Menber";
```
Comment (VinchC) [inline client/src/main.jsx]: Member/Member

File: client/src/pages/Member/Menber.jsx [hunk 12]
```diff
@@ -1 +1,12 @@
 import "./menber.css";
+
+
+function Menber() {
+    return (
+      <div>
+        <h1>Menber</h1>
```
Comment (VinchC) [inline client/src/pages/Member/Menber.jsx]: Member

File: client/src/pages/Member/Menber.jsx [hunk 13]
```diff
@@ -1 +1,12 @@
 import "./menber.css";
+
+
+function Menber() {
```
Comment (VinchC) [inline client/src/pages/Member/Menber.jsx]: Member

File: client/src/pages/Member/Menber.jsx [hunk 14]
```diff
@@ -1 +1,12 @@
...
+    return (
+      <div>
+        <h1>Menber</h1>
+      </div>
+    );
+  }
+  
+  export default Menber;
```
Comment (VinchC) [inline client/src/pages/Member/Menber.jsx]: Member

File: server/database/schema.sql [hunk 15]
```diff
@@ -1,46 +1,116 @@
...
+-- Schema mydb
+-- -----------------------------------------------------
+
+-- -----------------------------------------------------
+-- Table `User`
+-- -----------------------------------------------------
+CREATE TABLE IF NOT EXISTS `User` (
+  `id` INT NOT NULL,
+  `username` VARCHAR(50) NULL,
+  `Email` VARCHAR(100) NULL,
```
Comment (VinchC) [inline server/database/schema.sql]: email

File: server/database/schema.sql [hunk 16]
```diff
@@ -1,46 +1,116 @@
...
+  `password` VARCHAR(80) NULL,
+  `Is_admin` TINYINT NULL,
```
Comment (VinchC) [inline server/database/schema.sql]: is_admin NOT NULL default 0

File: server/database/schema.sql [hunk 17]
```diff
@@ -1,46 +1,116 @@
...
+-- -----------------------------------------------------
+
+-- -----------------------------------------------------
+-- Table `User`
+-- -----------------------------------------------------
+CREATE TABLE IF NOT EXISTS `User` (
+  `id` INT NOT NULL,
+  `username` VARCHAR(50) NULL,
+  `Email` VARCHAR(100) NULL,
+  `password` VARCHAR(80) NULL,
```
Comment (VinchC) [inline server/database/schema.sql]: `password` VARCHAR(255) NOT NULL

File: server/database/schema.sql [hunk 18]
```diff
@@ -1,46 +1,116 @@
...
+-- -----------------------------------------------------
+-- Schema mydb
+-- -----------------------------------------------------
+
+-- -----------------------------------------------------
+-- Table `User`
+-- -----------------------------------------------------
+CREATE TABLE IF NOT EXISTS `User` (
+  `id` INT NOT NULL,
+  `username` VARCHAR(50) NULL,
```
Comment (VinchC) [inline server/database/schema.sql]: NOT NULL

File: server/database/schema.sql (hunk 15 above)
Comment (VinchC) [inline server/database/schema.sql]: NOT NULL

File: server/database/schema.sql [hunk 19]
```diff
@@ -1,46 +1,116 @@
...
+  PRIMARY KEY (`id`))
+ENGINE = InnoDB;
+
+
+-- -----------------------------------------------------
+-- Table `Team`
+-- -----------------------------------------------------
+CREATE TABLE IF NOT EXISTS `Team` (
+  `id` INT NOT NULL,
+  `Team_Name` VARCHAR(55) NULL,
```
Comment (VinchC) [inline server/database/schema.sql]: `team_name` NOT NULL

File: server/database/schema.sql [hunk 20]
```diff
@@ -1,46 +1,116 @@
...
+  PRIMARY KEY (`id`))
+ENGINE = InnoDB;
+
+
+-- -----------------------------------------------------
+-- Table `Project`
+-- -----------------------------------------------------
+CREATE TABLE IF NOT EXISTS `Project` (
+  `id` INT NOT NULL,
```
Comment (VinchC) [inline server/database/schema.sql:38]: all field names in lower case letter and check nullable property of fields - NOT NULL should be the majority

**Review by VinchC — CHANGES_REQUESTED** [review]

**Review by VinchC — APPROVED** [review]

**Pull request merged**